tests/forms_tests/tests/test_formsets.py

```diff
@@ -591,6 +592,31 @@ def test_formsets_with_ordering(self):
             ],
         )
 
+    def test_formsets_with_order_custom_widget(self):
+        class OrderingAttributFormSet(BaseFormSet):
+            ordering_widget = HiddenInput
+
+        class OrderingMethodFormSet(BaseFormSet):
+            def get_ordering_widget(self):
+                return HiddenInput(attrs={'class': 'ordering'})
+
+        tests = (
+            (OrderingAttributFormSet, '<input type="hidden" name="form-0-ORDER">'),
+            (OrderingMethodFormSet, '<input class="ordering" type="hidden" name="form-0-ORDER">'),
+        )
+        for formset_class, order_html in tests:
+            with self.subTest(formset_class=formset_class):
```

Comment:
hi @felixxm @hramezani , subTest parameter is the internal class. Because of that, when the test is failed, gets an error **"AttributeError: Can't pickle local object 'FormsFormsetTestCase.test_formsets_with_order_custom_widget.<locals>.OrderingMethodFormSet'"**. 

Should it be changed as `with self.subTest(order_html=order_html):`?

Reply:
Use `runtests.py --parallel=1` to avoid that crash. See the warning in the [--parallel](https://docs.djangoproject.com/en/dev/ref/django-admin/#cmdoption-test-parallel) docs

Reply:
Ah, good. Widgets... I think something like `formset_class=formset_class.__name__` would be clearer than the HTML string. 

Then at least you'd get this:

```
FAIL: test_formsets_with_order_custom_widget (tests.forms_tests.tests.test_formsets.FormsFormsetTestCase) [<object object at 0x10456f0a0>] (formset_class='OrderingMethodFormSet')
```

... which clearly tells you which case went wrong.

Reply:
+1

Reply:
> Use `runtests.py --parallel=1` to avoid that crash. See the warning in the [--parallel](https://docs.djangoproject.com/en/dev/ref/django-admin/#cmdoption-test-parallel) docs

thanks. I didn't know this.